azurerm/resource_arm_role_assignment.go

```diff
@@ -64,6 +65,24 @@ func resourceArmRoleAssignment() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+
+			"principal_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(authorization.Application),
+					string(authorization.DirectoryObjectOrGroup),
+					string(authorization.DirectoryRoleTemplate),
+					string(authorization.Everyone),
+					string(authorization.ForeignGroup),
+					string(authorization.Group),
+					string(authorization.MSI),
+					string(authorization.ServicePrincipal),
+					string(authorization.Unknown),
```

Comment:
presumably `Unknown` shouldn't be in this list?

Reply:
just because it's in the list supported by the API doesn't mean it should be exposed to users unfortunately, Unknown is generally present to indicate a value that's unsupported on this API version (for example if it's been created on a newer/older version which is no longer supported)

Reply:
@tombuildsstuff I have asked the service team about the expected behavior and the purpose of the `Unknown` value.